test/js/node/child_process/child_process.test.ts

```diff
@@ -214,6 +215,19 @@ describe("spawn()", () => {
     const { stdout } = spawnSync("echo", ["hello"], { encoding: "utf8" });
     expect(stdout.trim()).toBe("hello");
   });
+
+  test.if(isLinux)("should handle null for first 4 positions of stdio", async () => {
+    // Run a program that does nothing except pause indefinitely, thus an
+    // opportunity to inspect its file descriptors.
+    const proc = spawn("sleep", ["inf"], { stdio: [null, null, null, null] });
+    // Delay here because the command (above) opens and closes a bunch of
+    // files when starting up, so we have to wait for it to stabilize.  This
+    // still is a race condition though, would be nice to eliminate.
+    await Bun.sleep(10);
```

Comment:
Love the comments, too! This looks great.

I think it's sufficient to have a comment here about the existence of the race condition, which I suspect you could hit:
1. running on a single vCPU
2. in QEMU
3. while several other processes are running on the VM
4. with a guest kernel built with HZ=100

If you do want to eliminate it, we can wait for /proc/${pid}/status to include "State:\tS", which would indicate the dynamic loader has done its job and the process is now in sleep state, rather than running or in disk sleep. We could also use "yes" and wait for the first line of output, at which point the loader should have closed its fds, but that relies on a specific implementation of "yes"...

Reply:
The "yes" proposal seems pretty good to me; A very creative idea. Is there a sensible scenario where it wouldn't work? I'm failing to think of one myself.

Reply:
> The "yes" proposal seems pretty good to me; A very creative idea. Is there a sensible scenario where it wouldn't work? I'm failing to think of one myself.

Well, I'm sure people have rewritten "yes" in a number of programming languages, and some of them might open more fds. I don't think that'd be a sensible thing to do, though... probably not worth worrying about

Reply:
Updated comment and now using yes. Hopefully it's a net improvement? (Feedback welcome, of course.)

Reply:
I did also test w/ and w/o the original regression (the typo) to make sure the test is still effective with "yes".